mkosi/distributions/gentoo.py

```diff
@@ -211,125 +118,142 @@ def fetch_fix_stage3(self) -> None:
+                state.cache_dir.joinpath(d).mkdir(exist_ok=True)
+
+            config = state.pkgmngr / "etc/portage"
+
+            package_use = config / "package.use"
+            package_use.mkdir(parents=True, exist_ok=True)
+            package_use.joinpath("systemd").write_text(
+                # repart for usronly
+                dedent(
+                    """\
+                    # MKOSI: used during the image creation
+                    # "/usr/lib/systemd/boot/efi": No such file or directory
+                    sys-apps/systemd gnuefi
+                    sys-apps/systemd -cgroup-hybrid
+                    sys-apps/systemd elfutils # for coredump
+
+                    sys-apps/systemd homed cryptsetup -pkcs11
+                    # See: https://bugs.gentoo.org/832167
+                    sys-auth/pambase homed
+
+                    # MKOSI: usronly
+                    sys-apps/systemd repart
+                    # MKOSI: make sure we're init (no openrc)
+                    sys-apps/systemd sysv-utils
+                    """
+                )
             )
-        )
 
-    def mkosi_conf(self) -> None:
-        package_env = self.portage_cfg_dir / "package.env"
-        package_env.mkdir(exist_ok=True)
-        self.ebuild_sh_env_dir.mkdir(exist_ok=True)
-
-        # apply whatever we put in mkosi_conf to runs invocation of emerge
-        package_env.joinpath("mkosi.conf").write_text("*/*    mkosi.conf\n")
-
-        # we use this so we don't need to touch upstream files.
-        # we also use this for documenting build environment.
-        emerge_vars_str = ""
-        emerge_vars_str += "\n".join(f'{k}="${{{k}}} {v}"' for k, v in self.emerge_vars.items())
-
-        self.ebuild_sh_env_dir.joinpath("mkosi.conf").write_text(
-            dedent(
-                f"""\
-                # MKOSI: these were used during image creation...
-                # and some more! see under package.*/
-                {emerge_vars_str}
-                """
+            package_env = config / "package.env"
+            package_env.mkdir(parents=True, exist_ok=True)
+            # apply whatever we put in mkosi_conf to runs invocation of emerge
+            package_env.joinpath("mkosi.conf").write_text("*/* mkosi.conf\n")
+
+            # we use this so we don't need to touch upstream files.
+            # we also use this for documenting build environment.
+            ebuild_sh_env_dir = config / "env"
+            ebuild_sh_env_dir.mkdir(parents=True, exist_ok=True)
+            emerge_vars_str = ""
+            emerge_vars_str += "\n".join(f'{k}="${{{k}}} {v}"' for k, v in emerge_vars.items())
+            ebuild_sh_env_dir.joinpath("mkosi.conf").write_text(
+                dedent(
+                    f"""\
+                    # MKOSI: these were used during image creation...
+                    # and some more! see under package.*/
+                    {emerge_vars_str}
+                    """
+                )
```

Comment:
The emerge vars make sense when emerge is invoked by mkosi but I don't think they make sense to use in the image as well. Again, this should probably be left to the distro defaults or configured by the user but not configured by mkosi

Reply:
my idea was to document what we used for building the image. i can drop the documentation dance but we needed those for `emerge` to function properly in a sandboxed enviroment. this might have changed by now. that bit of code is old.

Reply:
... let me know if we want the documentaion bit to be dropped, please.

Reply:
Yeah let's drop it, we don't do it for other distros either